integration-tests/matching.ts

```diff
@@ -35,6 +36,194 @@ void test('Pupil Request Match', async () => {
+
+    // Creating a Chat ensures all users have accounts:
+
+    // The pupil does not have one, it gets created:
+    expectFetch({
+        method: "GET",
+        url: `https://api.talkjs.com/v1/mocked-talkjs-appid/users/pupil_${pupil.pupil.id}`,
+        responseStatus: 404
+    });
+
+    expectFetch({
+        url: `https://api.talkjs.com/v1/mocked-talkjs-appid/users/pupil_${pupil.pupil.id}`,
+        "method": "PUT",
+        "body": JSON.stringify({"name": `${pupil.firstname} ${pupil.lastname}`, "email": [pupil.email.toLowerCase()], "role": "pupil"}),
+        "responseStatus": 200
+    });
+
+    expectFetch({
+        method: "GET",
+        url: `https://api.talkjs.com/v1/mocked-talkjs-appid/users/pupil_${pupil.pupil.id}`,
+        responseStatus: 200,
+        response: {} // TODO: Mock properly
+    });
+
+
+    // The student has one:
+
+    expectFetch({
+        method: "GET",
+        url: `https://api.talkjs.com/v1/mocked-talkjs-appid/users/student_${student.student.id}`,
+        responseStatus: 200,
+        response: {} // TODO: Mock properly
+    });
+
+    // Then the conversion is created:
+
+    expectFetch({
+        url: "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*",
+        method: "GET",
+        responseStatus: 404
+      });
+
+        // TODO: Remove duplicate fetch
+      expectFetch({
+        url: "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*",
+        method: "GET",
+        responseStatus: 404
+      });
+
+      // TODO: Why PUT twice?
+      expectFetch({
+        url: "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*",
+        method: "PUT",
+        body: "{\"id\":\"*\",\"custom\":{\"type\":\"match\"}}",
+        responseStatus: 200
+      });
+
+      expectFetch({
+        url: "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*",
+        method: "PUT",
+        body: `{\"custom\":{\"type\":\"match\"},\"participants\":[\"pupil_${pupil.pupil.id}\",\"student_${student.student.id}\"]}`,
+        responseStatus: 200
+      });
+
+      expectFetch({
+        url: "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*",
+        method: "GET",
+        responseStatus: 200,
+        response: { "id": "mocked" } // TODO: mock propery
+      });
+
+      // TODO: Why twice?
+      expectFetch({
+        "url": "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*",
+        "method": "GET",
+        "responseStatus": 200,
+        "response": { "id": "mocked" } // TODO: mock properly
+      });
+
+      expectFetch({
+        "url": "https://api.talkjs.com/v1/mocked-talkjs-appid/conversations/*/messages",
+        "method": "POST",
+        "body": "[{\"text\":\"Willkommen im Lern-Fair Chat!\",\"type\":\"SystemMessage\",\"custom\":{\"type\":\"first\"}}]",
+        "responseStatus": 200
+      });
+
+
+    const { matchChatCreate: conversionID } = await pupilClient.request(`
```

Comment:
Should we check if the response make sense or is it enough that it didn't crash?